site/server/app/com/fortysevendeg/exercises/persistence/repositories/UserProgressRepository.scala

```diff
@@ -83,6 +71,24 @@ class UserProgressDoobieRepository(implicit persistence: PersistenceModule) exte
     Q.findBySection, (userId, libraryName, sectionName)
   )
 
+  override def findUserProgressBySection(
+    user:        User,
+    libraryName: String,
+    sectionName: String
+  ): ConnectionIO[SectionProgress] =
+    findBySection(user.id, libraryName, sectionName) map {
+      list ⇒
+        val exercisesList: List[LibrarySectionExercise] = list map { up ⇒
+          val argsList: List[String] = up.args map (_.split("##").toList) getOrElse Nil
```

Comment:
Please add a ticket to get args in exercises generalized to a better format than this.